exp/controllers/awsmachinepool_controller.go

```diff
@@ -145,7 +147,24 @@ func (r *AWSMachinePoolReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 		return ctrl.Result{}, errors.Errorf("failed to create scope: %+v", err)
 	}
 
-	// todo: defer conditions + machinePoolScope.Close()
+	// Always close the scope when exiting this function so we can persist any AWSMachine changes.
+	defer func() {
+		// set Ready condition before AWSMachine is patched
+		conditions.SetSummary(machinePoolScope.AWSMachinePool,
+			conditions.WithConditions(
+				infrav1.InstanceReadyCondition,
+				infrav1.SecurityGroupsReadyCondition,
```

Comment:
i'm not sure if either of these conditions apply to ASGs - we may want to define some new ones in expinfrav1

Reply:
unless InstanceReadyCondition is part of the contract

Reply:
I am fairly sure it was a const that was defined in that status struct. I can make it more relevant. Was unsure how much I was allowed to deviate. 
I will resolve this in the next commit.